src/parser.ts

```diff
@@ -0,0 +1,37 @@
+import semver from "semver";
+import {
+  PackageJsonType,
+  PackageInfoBase,
+  DependencyObjectType,
+  packageJsonSchema,
+} from "./types";
+import * as fs from "fs";
+import * as path from "path";
+
+const normalizeVersion = (version: string): string =>
+  semver.coerce(version)?.version ?? "";
+
+export const readJsonFile = (pathToPackage: string): PackageJsonType => {
+  const fileContent = fs.readFileSync(
+    path.join(__dirname, pathToPackage),
+    "utf8"
+  );
+  return packageJsonSchema.parse(JSON.parse(fileContent));
+};
+
+const parseDependencyArray = (
+  dependencies: DependencyObjectType,
+  isDevDependency: boolean
+): PackageInfoBase[] =>
+  Object.entries(dependencies).map(([name, version]) => ({
```

Comment:
i think we should change this. i didn't think it was very clean when i was writing it

Reply:
but because our data format is {string:string} we cant really do dependencies.map since its not an array so no sure hwo to make it cleaner

Reply:
Yeah not sure how we could improve it yet, but open to ideas!